pkg/systemd/services/service-details.jsx

```diff
@@ -208,11 +208,40 @@ export class ServiceDetails extends React.Component {
             waitsFileAction: false,
             note: "",
             error: "",
+            delay: 30000, // 30s
+            unit_properties: {},
         };
 
         this.onOnOffSwitch = this.onOnOffSwitch.bind(this);
         this.unitAction = this.unitAction.bind(this);
         this.unitFileAction = this.unitFileAction.bind(this);
+        this.addUnitProperties = this.addUnitProperties.bind(this);
+        this.doPropertyPolling = this.doPropertyPolling.bind(this);
+
+        this.unitType = props.unit.Id.split('.').slice(-1)[0];
+        this.unitTypeCapitalized = this.unitType.charAt(0).toUpperCase() + this.unitType.slice(1);
+
+        if (props.unit.ActiveState == "active") {
+            this.doPropertyPolling();
+            this.interval = setInterval(this.doPropertyPolling, this.state.delay);
+        }
+    }
+
+    componentDidUpdate(prevProps) {
+        // If unit became active start property polling and if got inactive stop
+        if (this.props.unit.ActiveState === 'active' && !this.interval) {
+            this.doPropertyPolling();
+            clearInterval(this.interval);
+            this.interval = setInterval(this.doPropertyPolling, this.state.delay);
+        }
+        if (this.props.unit.ActiveState === 'inactive' && this.interval) {
+            this.doPropertyPolling();
+            clearInterval(this.interval);
+        }
+    }
+
+    componentWillUnmount() {
+        clearInterval(this.interval);
```

Comment:
Does this cause any error message when this.interval is invalid? To be on the safe side, this could explicitly check.

Reply:
Seems, not but guarded it anyway.